src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+        Latest_End_Date__c from Address__c];
+        system.assertEquals(4, ListAddr.size());
+        for (Address__c addr : ListAddr) {
+            boolean fNewDefault = (addr.MailingStreet__c.contains('New Default Street'));
+            system.assertEquals(fNewDefault, addr.Default_Address__c);
+            if (fNewDefault) {
+               system.assertEquals(system.today(), addr.Latest_Start_Date__c);
+               system.assertEquals(null, addr.Latest_End_Date__c);
+            } else {
+               system.assertEquals(system.today(), addr.Latest_End_Date__c);                
+            }
+        }        
+    }
+    
+    /*********************************************************************************************************
+    @description Insert new default addresses to Contact with existing default addresses, and then go back  
+    verify:
+        Contact address matches original default
+        new default addresses no longer marked default
+    **********************************************************************************************************/            
+    @isTest
+    public static void insertNewDefaultAddrAndReset() {
+        configSettings();
+        
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+        List<Address__c> ListAddrOriginal = contactsAddrs.addrs;
+        
+        // create additional addresses
+        List<Address__c> addrs = UTIL_UnitTestData_TEST.initTestAddr(2);
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            addrs[i].Parent_Contact__c = contactsAddrs.contacts[i].Id;
+            addrs[i].Default_Address__c = true;
+            addrs[i].MailingStreet__c = 'New Default Street' + i;
+            addrs[i].MailingCity__c = 'New Default City' + i;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        insert addrs;
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        
+        // go back to original default addresses
+        // added this extra set to test fix where we didn't use the correct default
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            ListAddrOriginal[i].Default_Address__c = true;
+            ListAddrOriginal[i].MailingStreet__c = 'Original Default Street' + i;
+            ListAddrOriginal[i].MailingCity__c = 'Original Default City' + i;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update ListAddrOriginal;
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.stopTest();
+        
+        // verify the additional addresses got Default cleared.
+        // and verify latest start date and latest end date appropriately set.
+        List<Address__c> ListAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Contact__c, Latest_Start_Date__c, 
```

Comment:
Variables should use the `lowerCamelCase` naming convention. Please rename accordingly.